framework/src/controller/application.js

```diff
@@ -119,6 +124,11 @@ class Application {
 		__private.modules.set(this, {});
 		__private.transactions.set(this, {});
 
+		this.applicationState = new ApplicationState(
```

Comment:
considering multi-process application, ApplicationState need to emit/subscribe to event, and not passing instance to other modules. Please talk to @jondubois and @lsilvs regards to this